server/controllers/admin.js

```diff
@@ -0,0 +1,215 @@
+          user: result[1][0],
+        });
+      }
+      return res.status(200).json({
+        status: 200,
+        user: result[1][0],
+      });
+    } catch (error) {
+      Admin.errorHandler(res, error);
+    }
+  }
+
+  /**
+   * The controller to update a user by the admin.
+   * @param {req} req the request.
+   * @param {res} res the response.
+   * @returns {void}
+  */
+  static async deleteUsers(req, res) {
+    const { username } = req.params;
+    const user = await Users.getUser(username, res);
+    if (!user) {
+      return res.status(404).json({
+        status: 404,
+        message: 'User not found',
+      });
+    }
+    await User.destroy({ where: { username, }, returning: true });
+    return res.status(200).json({
+      status: 200,
+      message: `User ${user.get().username} deleted successfully`,
+    });
+  }
+
+  /**
+   * The controller to update a user by the admin.
+   * @param {req} req the request.
+   * @param {res} res the response.
+   * @returns {void}
+  */
+  static async makeAdmin(req, res) {
+    const { username } = req.params;
+    const user = await Users.getUser(username, res);
+    if (!user) {
+      return res.status(404).json({
+        status: 404,
+        message: 'User not found',
+      });
+    }
+    if (user.get().roles === 'admin') {
+      return res.status(400).json({
+        status: 400,
+        message: `${username} is already an admin`,
+      });
+    }
+    const newAdmin = await User.update(
+      { roles: 'admin', },
+      {
+        where: { username },
+        returning: true
+      }
+    );
+    delete newAdmin[1][0].get().password;
+    return res.status(200).json({
+      status: 200,
+      message: `${username} is now an admin`,
+      user: newAdmin[1][0],
+    });
+  }
+
+
+  /**
+   * The controller to update a user by the admin.
+   * @param {req} req the request.
+   * @param {res} res the response.
+   * @returns {void}
+  */
+  static async makeUser(req, res) {
+    const { username } = req.params;
+    const user = await Users.getUser(username, res);
+    if (!user) {
+      return res.status(404).json({
+        status: 404,
+        message: 'User not found',
+      });
+    }
+    if (user.get().roles === 'user') {
+      return res.status(400).json({
+        status: 400,
+        message: `${username} is already a normal user`,
+      });
+    }
+    const newAdmin = await User.update(
+      { roles: 'user', },
+      {
+        where: { username },
+        returning: true
+      }
+    );
+    delete newAdmin[1][0].get().password;
+    return res.status(200).json({
+      status: 200,
+      message: `${username} is now a normal user`,
+      user: newAdmin[1][0],
+    });
+  }
+
+  /**
+   * The controller to update a user by the admin.
+   * @param {string} res the request.
+   * @param {string} error the response.
+   * @returns {void}
+  */
+  static errorHandler(res, error) {
```

Comment:
Hey @dusmel, would you please put this errorHandler function in /helpers/errorHandler.js for consistency? Also, try to correct the JSDoc of it. Thanks

Reply:
yeah it makes sense 👍